src/main/java/pages/HomePage.java

```diff
@@ -47,6 +48,15 @@ public void searchText(String textForSearch) {
         searchButton.click();
     }
 
+    public void clickOnAmazonLogo() {
+        amazonLogo.click();
+    }
+
+    public void clickOnTodaySDeals(String text) {
+        clickOnAmazonLogo();
+        driver.findElement(By.linkText(text)).click();
```

Comment:
1. Why are you clicking on amazon logo?
2. If you know exact what element is going to be clicked, why you pass a text to the method? Your method name already says what you're going to do. If you want to click by linktext, then change the method name, so it will reflect its true function.